.github/workflows/cron-update-rust.yml

```
@@ -0,0 +1,43 @@
+name: Update rust version
+on:
+  schedule:
+    - cron: "0 0 15 * *" # At 00:00 on day-of-month 15.
+  workflow_dispatch: # allows manual triggering
+jobs:
+  format:
+    name: Update rustc
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@stable
+      - name: Update rust-version to use latest stable
+        run: |
+          set -x
+          # Extract the version from whatever version of the compiler dtolnay/rust-toolchain gives us.
+          RUST_VERSION=$(rustc --verbose --version | sed -ne 's/^release: //p')
+          # Update the version in the reference file.
+          echo "${RUST_VERSION}" > rust-version
+          echo "rust_version=${RUST_VERSION}" >> $GITHUB_ENV
+          # In case of no new version don't make an empty PR.
+          if ! git diff --exit-code > /dev/null; then
+              echo "Updated rustc. Opening PR."
+              echo "changes_made=true" >> $GITHUB_ENV
+          else
+              echo "Attempted to update rustc but the latest stable date did not change. Not opening any PR."
+              echo "changes_made=false" >> $GITHUB_ENV
+          fi
+      - name: Create Pull Request
+        if: env.changes_made == 'true'
+        uses: peter-evans/create-pull-request@v6
+        with:
+          # TODO: Allow `MY_TOKEN` to open a pull request
+          # token: ${{ secrets.MY_TOKEN }}
```

I think we don't need a custom token and can use [GITHUB_TOKEN](https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token) instead.

```suggestion
          # token: ${{ secrets.GITHUB_TOKEN }}
```

---

Ok changed to `GITHUB_TOKEN`. As far as permissions I guess it needs to trigger other workflows in addition to creating a PR. We should be able to manually run the cron update to check that it's working